src/CoaddBoundedField.cc

```diff
@@ -239,6 +239,18 @@ PTR(afw::math::BoundedField) CoaddBoundedField::operator*(double const scale) co
     throw LSST_EXCEPT(pex::exceptions::NotFoundError, "Scaling of CoaddBoundedField is not implemented");
 }
 
+bool CoaddBoundedField::operator==(BoundedField const& rhs) const {
+    try {
+        auto& rhsCasted = dynamic_cast<CoaddBoundedField const&>(rhs);
+        return (getBBox() == rhsCasted.getBBox()) && (_default == rhsCasted._default) &&
+               ((*_coaddWcs) == (*rhsCasted._coaddWcs)) && (_elements == rhsCasted._elements);
+    }
+    catch (std::bad_cast&) {
+        return false;
+    }
```

Comment:
I'd recommend converting the try block to:
```
auto rhsCasted = dynamic_cast<CoaddBoundedfield const *>(&rhs);
if (!rhsCasted) {
    return false;
}
...
```
Some (most?) compilers make checking for `nullptr` *much* faster than throwing and catching an exception, so when you don't actually gain anything from exceptions (e.g. unwinding the stack gracefully) it's better to use the pointer form of `dynamic_cast`.